modules/axtask/src/run_queue.rs

```diff
@@ -438,6 +439,22 @@ impl AxRunQueue {
             return;
         }
 
+        // Task must be scheduled atomically, wait for next task's scheduling process to complete.
+        // If the owning (remote) CPU is still in the middle of schedule() with
+        // this task (next task) as prev, wait until it's done referencing the task.
+        //
+        // Pairs with the `clear_prev_task_on_cpu()`.
+        //
+        // Note:
+        // 1. This should be placed after the judgement of `TaskState::Blocked,`,
+        //    because the task may have been woken up by other cores.
+        // 2. This can be placed in the front of `switch_to()`
+        #[cfg(feature = "smp")]
+        while next_task.on_cpu() {
```

Comment:
Why put it here?

Reply:
because of `migrate_current`,
we will put current task into other CPU's run queue and set it as ready, where it maybe be scheduled on other cores.

it faces the same **READ-AFTER-WRITE** problem, we have make the scheduling process atomically.

Reply:
Can we reimplement `migrate_current` with `unblock_task` and `resched`? And keep the `on_cpu` check loop in `unblock_task`.

Reply:
No, I do not think so.
`unblock_task` is task A try to unblock task B, where task A may loop on checking the `on_cpu` flag of task B.

while `migrate_current` is another situation where task A try to migrate itself (or you can treat it as unblock itself on other RQs), a task can never loop on the `on_cpu` flag of itself, right?